Build.bee.cs

```diff
@@ -0,0 +1,377 @@
+using Bee.Toolchain.VisualStudio;
+using Bee.Toolchain.Windows;
+using Bee.Tools;
+using NiceIO;
+
+class Build
+{
+    static void Main()
+    {
+        using var _ = new Bee.Core.BuildProgramContext();
+
+        Backend.Current.StevedoreSettings = new StevedoreSettings
+        {
+            Manifest = { "manifest.stevedore" },
+            EnforceManifest = true,
+        };
+
+        foreach (var sdk in Sdks.ForCurrentPlatformEditorSdks())
+        {
+            foreach (CodeGen codeGen in Enum.GetValues(typeof(CodeGen)))
+            {
+                var programConfiguration = new ProgramConfiguration(sdk, codeGen);
+                SetupLib(programConfiguration);
+                SetupStevedoreArtifact(programConfiguration);
+            }
+        }
+    }
+
+    enum CodeGen
+    {
+        Debug,
+        Release
+    }
+
+    class ProgramConfiguration
+    {
+        public ProgramConfiguration(Sdk sdk, CodeGen codeGen)
+        {
+            Sdk = sdk;
+            CodeGen = codeGen;
+        }
+
+        public string GetTargetName(string separator = "_")
+        {
+            var targetNameParts = new List<string> {
+                Sdk.Platform.DisplayName.ToLower(),
+                Sdk.Architecture.DisplayName
+            };
+
+            if (CodeGen == CodeGen.Debug)
+            {
+                targetNameParts.Add("dbg");
+            }
+
+            return string.Join(separator, targetNameParts);
+        }
+
+        public string GetArtifactName(string separator = "_")
+        {
+            var artifactNameParts = new List<string> {
+                "tbb",
+                GetTargetName(separator: separator)
+            };
+
+            return string.Join(separator, artifactNameParts);
+        }
+
+        public Sdk Sdk { get; private set; }
+        public CodeGen CodeGen { get; private set; }
+    }
+
+    static void SetupLib(ProgramConfiguration programConfiguration)
+    {
+        var cfg = CfgFor(programConfiguration.CodeGen);
+        var targetId = programConfiguration.GetTargetName();
+
+        NPath[] inputs;
+        var commandLineArguments = new List<string>
+        {
+            $"arch={ArchFor(programConfiguration.Sdk.Architecture)}",
+            $"compiler={CompilerFor(programConfiguration.Sdk)}",
+            $"cfg={cfg}",
+            $"tbb_build_prefix={targetId}",
+            $"-j {Environment.ProcessorCount}",
+        };
+        Dictionary<string, string> environmentVariables;
+
+        var path = Environment.GetEnvironmentVariable("PATH")?.Split(Path.PathSeparator).ToNPaths().ToArray() ?? Array.Empty<NPath>();
+    
+        if (programConfiguration.Sdk is WindowsSdk)
+        {
+            WindowsSdk windowsSdk = (WindowsSdk)programConfiguration.Sdk;
+
+            var binPaths = windowsSdk.EnvironmentVariables["PATH"].Split(";").ToNPaths();
+            inputs = binPaths
+                .Concat(windowsSdk.IncludePaths)
+                .Concat(windowsSdk.LibraryPaths)
```

Comment:
Why are you adding include and library directories to the PATH? Are there executables in there?

Reply:
They're not being added to the PATH, they're being added as inputs for the make bee action. The VisualStudioSdk class doesn't expose the Windows SDK and MSVC Toolset artifacts directly so I need to add paths from within both as inputs to ensure they're downloaded, although it's not bad to be explicit about what parts of the artifacts we're using.

Reply:
Oh, right.